.github/workflows/main.yml

```diff
@@ -26,30 +26,25 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python: [3.7, 3.9]
+        python: [3.7, 3.8, 3.9]
 
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python 3.x
       uses: actions/setup-python@v2
       with:
         python-version:  ${{ matrix.python }}
+
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-    - name: Run tests
-      run: |
-        python -m unittest discover tests -v
-    - name: Generate coverage report
-      run: |
-        pip install pytest
-        pip install pytest-cov
-        pytest --cov-config=.coveragerc
-        pytest --cov=./ --cov-report=xml
+        pip install -r requirements.txt
+
+    - name: Run tests and generate report
+      run: coverage run -m unittest discover tests -v
+
     - name: Upload coverage to Codecov
       uses: codecov/codecov-action@v1
       with:
-        file: ./coverage.xml
         name: codecov-umbrella
```

Comment:
Probably not related to this issue but upload coverage only the the maxtrix version is 3.7. Can you make this change too ?

Reply:
> Do you want to upload coverage only when python version is 3.7?
> If yes, then that can be done with a if block but why we need that?

Because it's uploading the coverage for both the versions.

Reply:
I think we should maintain coverage for all versions. May be in future we have some compatibility shims then coverage will change for versions

Reply:
I don't see how the python version is going to affect the coverage : /.  Plus When the bot comments there is no way to differentiate between which coverage corresponds to which version.

Reply:
That was happening because all coverages have same name. Btw, I add a condition to upload coverage for Python 3.7